ctapipe/image/extractor.py

```diff
@@ -21,48 +21,60 @@
 import numpy as np
 from traitlets import Int
 from ctapipe.core import Component
-from numba import njit, prange, float64, float32, int64
-
-
-def extract_charge_from_peakpos_array(waveforms, peakpos, width, shift):
+from numba import njit, prange, guvectorize, float64, float32, int64
+
+
+@guvectorize(
+    [
+        (float64[:], int64, int64, int64, float64[:]),
+        (float32[:], int64, int64, int64, float64[:]),
+    ],
+    '(s),(),(),()->()',
+    nopython=True,
+)
+def sum_samples_around_peakpos(waveforms, peakpos, width, shift, ret):
```

Comment:
isn't `peakpos` now renamed `pulse_time`? perhaps updating the terminology everywhere is useful (just doing a replace all should work)

Reply:
Ah, ok.  Perhaps `peak_index` is better?  it's not clear from the name that it is different from the peak time

Reply:
Okay, I have switched to `peak_index`